suspected panic when error in closing websocket

```
2024-08-27 08:19:07.848	
rollapp.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
2024-08-27 08:19:07.715	
	/home/ubuntu/go/pkg/mod/github.com/dymensionxyz/dymint@v1.2.0-rc03/rpc/json/service.go:137 +0x30c



2024-08-27 08:19:07.715	
created by github.com/dymensionxyz/dymint/rpc/json.(*service).Subscribe in goroutine 5390
2024-08-27 08:19:07.715	
	/home/ubuntu/go/pkg/mod/github.com/dymensionxyz/dymint@v1.2.0-rc03/rpc/json/service.go:156 +0x244
2024-08-27 08:19:07.715	
github.com/dymensionxyz/dymint/rpc/json.(*service).Subscribe.func1({0x40352b2370, 0x3, 0x4003f79140?})
2024-08-27 08:19:07.715	
goroutine 18442 [running]:
2024-08-27 08:19:07.715	
panic: send on closed channel
2024-08-27 08:19:07.711	
time="2024-08-27T06:19:07Z" level=info msg="Failed to convert subscriptionID to int[]"
2024-08-27 08:19:07.711	
time="2024-08-27T06:19:07Z" level=info msg="Failed to convert subscriptionID to int[]"
2024-08-27 08:19:07.711	
time="2024-08-27T06:19:07Z" level=info msg="Failed to convert subscriptionID to int[]"
2024-08-27 08:19:07.711	
time="2024-08-27T06:19:07Z" level=info msg="Failed to convert subscriptionID to int[]"
```


i think its caused by this error closing websocket [https://grafana.testnets.noisnemyd.xyz/explore?schemaVersion=1&panes=%7B%2233d%22:%7B%22data[…]isualisationType%22:%22logs%22%7D%7D%7D%7D%7D&orgId=1](https://grafana.testnets.noisnemyd.xyz/explore?schemaVersion=1&panes=%7B%2233d%22:%7B%22datasource%22:%22fdlwcuq623zeoa%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Brollapp%3D%5C%22allinbets_777777-1%5C%22,%20host%3D%5C%22ra-blumbus-allinbets-wasm-sequencer%5C%22%7D%20%7C~%20%60read:%20connection%20reset%20by%20peer%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22fdlwcuq623zeoa%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%22now-6h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22logs%22:%7B%22visualisationType%22:%22logs%22%7D%7D%7D%7D%7D&orgId=1)
[9:02](https://dymensionworkspace.slack.com/archives/C07JJK96PT7/p1724742147458949)
and crashes here because channel closed https://github.com/dymensionxyz/dymint/blob/9aff932778375f3f5548b878ced85c8211300aec/rpc/json/service.go#L156

--- a/rpc/json/ws.go
+++ b/rpc/json/ws.go
@@ -8,7 +8,6 @@ import (
 	"time"
 
 	"github.com/gorilla/websocket"
-
 	tmpubsub "github.com/tendermint/tendermint/libs/pubsub"
 
 	"github.com/dymensionxyz/dymint/types"
@@ -79,12 +78,12 @@ func (h *handler) wsHandler(w http.ResponseWriter, r *http.Request) {
 		if err != nil {
 			if _, ok := err.(*websocket.CloseError); ok {
 				h.logger.Debug("WebSocket connection closed", "reason", err)
-				err := h.srv.client.EventBus.UnsubscribeAll(r.Context(), remoteAddr)
-				if err != nil && !errors.Is(err, tmpubsub.ErrSubscriptionNotFound) {
-					h.logger.Error("unsubscribe addr from events", "addr", remoteAddr, "err", err)
-				}
 			} else {
 				h.logger.Error("read next WebSocket message", "error", err)
+			}
+			err := h.srv.client.EventBus.UnsubscribeAll(r.Context(), remoteAddr)
+			if err != nil && !errors.Is(err, tmpubsub.ErrSubscriptionNotFound) {
+				h.logger.Error("unsubscribe addr from events", "addr", remoteAddr, "err", err)
 			}
 			break
 		}

